test/integration/enumerate_databases.test.ts

```diff
@@ -0,0 +1,121 @@
+import { expect } from 'chai';
+
+import { AddUserOptions, MongoClient, MongoServerError } from '../../src';
+
+describe('listDatabases', function () {
+  let client: MongoClient;
+
+  beforeEach(async function () {
+    client = this.configuration.newClient();
+    await client.connect();
+  });
+
+  afterEach(async function () {
+    await client?.close();
+  });
+
+  it('should return an Array', async () => {
+    const dbInfo = await client.db().admin().listDatabases();
+    expect(Array.isArray(dbInfo.databases)).to.be.true;
+  });
+
+  it('should contain no duplicates', async () => {
+    const dbInfo = await client.db().admin().listDatabases();
+    const databaseNames = dbInfo.databases.map(({ name }) => name);
+    const databaseNamesSet = new Set(databaseNames);
+    expect(databaseNames).to.have.lengthOf(databaseNamesSet.size);
+  });
+
+  // TODO(NODE-3860): Create driver test variants that require AUTH enabled
+  describe('authorizedDatabases flag', function () {
+    const username = 'a';
+    const password = 'b';
+    const mockAuthorizedDb = 'enumerate_databases';
+    const mockAuthorizedCollection = 'enumerate_databases_collection';
+
+    let client: MongoClient;
+    let authorizedClient: MongoClient;
+
+    const authorizedUserOptions: AddUserOptions = {
+      roles: [{ role: 'read', db: mockAuthorizedDb }]
+    };
+
+    beforeEach(function () {
+      if (process.env.AUTH !== 'auth') {
+        this.currentTest.skipReason =
+          'TODO(NODE-3860): Create driver test variants that require AUTH enabled';
+        this.skip();
+      }
+    });
+
+    beforeEach(async function () {
+      // pass credentials from cluster_setup's mlaunch defaults
+      // TODO(NODE-3860): pass credentials instead based on environment variable
+      client = this.configuration.newClient({ auth: { username: 'user', password: 'password' } });
+      await client.connect();
+
+      await client
+        .db(mockAuthorizedDb)
+        .createCollection(mockAuthorizedCollection)
+        .catch(() => null);
+
+      await client.db('admin').addUser(username, password, authorizedUserOptions);
+
+      authorizedClient = this.configuration.newClient({
+        auth: { username: username, password: password }
+      });
+      await authorizedClient.connect();
+    });
+
+    afterEach(async function () {
+      await client?.db('admin').removeUser(username);
+      await client?.db(mockAuthorizedDb).dropDatabase();
+      await client?.close();
+      await authorizedClient?.close();
+    });
+
+    it('should list authorized databases with authorizedDatabases set to true', async function () {
```

Comment:
can we add scenarios for passing true/false flag explicitly into listDatabases for the case of the user that has the full listDatabases permissions and show that the returned list is not affected?

Reply:
done!